src/python/WMCore/Storage/Backends/GFAL2Impl.py

```diff
@@ -28,17 +28,35 @@ def __init__(self, stagein=False):
         self.removeCommand = self.setups % '. $JOBSTARTDIR/startup_environment.sh; printenv; date; gfal-rm -vvv -t 600 %s '
         self.copyCommand = self.setups % '. $JOBSTARTDIR/startup_environment.sh; printenv; date; gfal-copy -vvv -t 2400 -T 2400 -p %(checksum)s %(options)s %(source)s %(destination)s'
 
-    def createSourceName(self, protocol, pfn):
+
+    def createFinalPFN(self, protocol, pfn):
         """
-        _createSourceName_
-        GFAL2 uses file:/// urls
+        _createFinalPFN_
+        GFAL2 requires file:/// for any direction transfers
         """
         if pfn.startswith('file:'):
             return pfn
         elif os.path.isfile(pfn):
```

Comment:
I thought isfile() would work only for regular files, not for remote ones. Can you x-check this works as expected?
BTW, isn't your `elif pfn.startswith('/'):` line below doing exactly the same check as above?